// gitbook-plugin-abc2svg/index.js
var Abc2Svg = require('./abc2svg-node.js');

module.exports = {
    book: {
    },
    blocks: {
        abc: {
            process: function(blk) {

                    var abc_source = "";
                    var abc = {};

                    function User() {
                      this.new_page = "";
                      this.read_file = function(fn) {
                        return readFile(fn)
                      };
                      this.errmsg = function(msg, l, c) {
                        this.errtxt += msg + "\n"
                      };
                      this.img_out = function(str) {
                        //console.log(str);
                        this.new_page += str
                      };
                      this.page_format = true
                    }

                    var user = new User;
                    abc_source = "%%fullsvg true \n" + // so that svg is shown correctly in pdf print
                                  //"%%writefields T false \n" + // dont print title
                                  //"%%pagewidth " + 800 + " \n" +
                                  //"%%leftmargin " + 0 + " \n" + 
                                  blk.body;

                    abc = new Abc2Svg.Abc(user);
                    abc.tosvg("this_string_is_only_for_errors", abc_source);

                    return (user.new_page);

            }
        }
    }
};
